### ISSUE TITLE
We need a proper blacklist/whitelist feature for CWs and posts

### ISSUE THREAD

**Issue by foervraengd** [mastodon/mastodon#19532]
### Pitch

Instead of putting the responsibility of the poster to hide their content under a CW for any possible thing that **might** upset people, have the CW rely on a whitelist/blacklist system that individual users can customize to their own liking.

### Motivation

I have been active on and off on a mastodon instance, and occasionally i see posts about some heated discourse/disagreement between people of different instances, arguing over what people should and should not have to put under a content warning label. It might not sound like a big deal at first, but i can def see it being a cause of bad relationships between people that otherwise would maybe get along just fine. Its also something i have seen before on other platforms that lacked flexible tools for hiding/showing content (the early days of discord had this problem for example). I think this scenario can be prevented without needing more work from moderators etc - just add a whitelist/blacklist for cws and posts.

I do know that there is already a filter list, but from what i experience, that filter completely hides the post entirely - as if it didnt even exist. I think this is a bit too harsh of an action, especially if people just wanna filter spoilers for a movie or something. I’d like to see something in between a cw and a filtered post - still visible on my tl, with the option to show it if im still curious.

I am not tech-savvy at all (ive used html like three times in my life), but i guess that the Easiest way to implement this, without it requiring a massive restructuring etc, would be to introduce a whitelist/blacklist system that only expands/hides posts that match said list.

I think its great that there is a CW feature, but due to how people label their cws in all kinds of vague ways, it can sometimes be hard to know what is hidden under the post. The CW as it is right now is only useful if everyone use it the exact same way, but people dont do that for various reasons.

Sure, one could argue that this stuff isnt mastodon’s responsibility - that people just need to learn to get along with each other etc. But some people would use that argument against having a CW feature to begin with.

 I think this would benefit everyone who are active on mastodon instances, especially people who might have specific triggers and fears for niche things. Everyone will have their own opinions on what counts as a cw, and people will always disagree with each other on it, but it should be possible to avoid these scenarios by design. Obviously it wont completely avoid it, but i am certain that it will reduce most of it and improve cooperation between people. (for example, i dont want to hide my post under a CW for everyone just because i drew a scary spider, but ill gladly give the possibility to those who want to avoid it with the option to still check it out when they feel like it )

I think that, by giving the power of the individual to use a blacklist/whitelist system that can CW/filter posts w the hashtag or phrase on their timeline, more people can easier have a better social experience on mastodon and its instances, even when people have their own social rules when it comes to content warnings. Flexibility!
here is a screenshot from my tumblr dashboard as a visual example of a post that contains tags i have on my own blacklist:

![IMG_7895](https://user-images.githubusercontent.com/116970273/198847679-6d49e183-e135-4e3b-8fca-0037a014cad0.jpeg)

**Comment by sealestial** [comment]
omg! please. this a much better way of actually curating what we consume on social media.
sometimes we follow a creator specifically bc of others might find triggering.
Using tags on posts, and letting us use a blacklist/whitelist system to filter is way more useful, even more so bc I can use block one word/tag and not see it from every poster, instead of relying on the poster's judgment if that should or not have a CW. not only that, but is useful for not getting spoilers from shows, etc.

**Comment by sudomemorizor** [comment]
its so annoying seeing a bunch of cw'd posts just let me hide them
also they should make the tags separated from the actual post or give you the option to do so because i hate twitter tag formatting so much

**Comment by Gargron** [comment]
I think you willl probably enjoy the post filtering feature being introduced in the upcoming 4.0. Please check if maybe this request is already fulfilled by it.

**Comment by foervraengd** [comment]
> I think you willl probably enjoy the post filtering feature being introduced in the upcoming 4.0. Please check if maybe this request is already fulfilled by it.

i am completely new to gitHub, so where can i read about this upcoming 4.0?

**Comment by yoshimo** [comment]
https://github.com/mastodon/mastodon/blob/main/CHANGELOG.md

**Comment by foervraengd** [comment]
> https://github.com/mastodon/mastodon/blob/main/CHANGELOG.md

thank you!
The changelog list doesnt really specify how the “filter individual posts” will work, but if Gargon said i will enjoy it then i hope its something close to what i am suggesting.

**Comment by ClearlyClaire** [comment]
The new filtering system is described in #18058. I think it mostly satisfies your request, but maybe not entirely, as you cannot express something like “filter posts that contain that keyword but not that other keyword”.

As for filtering individual posts, it's a slightly different feature that allows you to hide a post you've already seen by adding it to one of your filters. The idea is that if a particular post upsets you for a personal reason, you can hide it without having to block/mute people.

**Comment by foervraengd** [comment]
> The new filtering system is described in #18058. I think it mostly satisfies your request, but maybe not entirely, as you cannot express something like “filter posts that contain that keyword but not that other keyword”.
> 
yeah i do think that having a whitelist(aka allowlist) would be the main thing i would love to see here. Because a lot of people blur posts that arent even explicit material to me. I still am glad to see there are works in a direction close to what i’m hoping for. But maybe its best to first make sure the new filter system works well before adding more stuff to it?

**Comment by ClearlyClaire** [comment]
I am not completely sure how an allow-list would work but that is something we could consider in the future. For now, though, this is not a priority.

**Comment by bnmng** [comment]
I am all for this.  Sometimes there are important messages to get out including calls for political action or calls to help in a crisis.  Some people don't want to see these messages and they have a right not to, but others might be interested.  The problem is it's a lot harder to get people's attention from behind a CW, even if it's something they might want to see

**Comment by noiob** [comment]
The new UI is nice but what I'm really missing is a checkbox that limits a filter to a CW. I sometimes wanna mute specific CWs but not the word itself

**Comment by alper** [comment]
> but due to how people label their cws in all kinds of vague ways

I think one onramp to more standardisation would be to allow hashtags in the CW field. As far as I can tell right now, if you put them in there, they are not treated as a hashtag.

See: https://github.com/mastodon/mastodon/issues/20534#issuecomment-1313557524

**Comment by trwnh** [comment]
> Sometimes there are important messages to get out including calls for political action or calls to help in a crisis. Some people don't want to see these messages and they have a right not to, but others might be interested. The problem is it's a lot harder to get people's attention from behind a CW, even if it's something they might want to see

not sure how this is related but CWs aren't about not seeing something, they're about providing context -- personally i'm likely to scroll past a post if i don't know what it's about. using a CW is actually a good way to get attention, if you use the right summary! 

if you don't want to see something, then what you probably want is filters and for people to make better use of keywords.
